contrib/release.sh

```diff
@@ -88,6 +88,14 @@ else
    ./contrib/build-linux/sdist/build.sh
 fi
 
+# create source tarball
+srctarball="Electrum-source-$VERSION.tar.gz"
```

Comment:
```suggestion
# create source-only tarball
srctarball="Electrum-sourceonly-$VERSION.tar.gz"
```
I mean, the other tarball is pretty-much a source tarball too.
This one has removed
- generated code, which is source
- bundled dependencies, which are source-only
- and ok, fair enough, compiled locale

I would rather this had a name that gives at least a hint to users that this is not what they want. Maybe let's call it `sourceonly` then...

Reply:
No objection to `sourceonly` from my end.

Reply:
Sure.

Reply:
It is not sufficient to modify this string, the file should be named as such in the first place :)

The sdist is created here:
https://github.com/spesmilo/electrum/blob/cbe1f3fd807b0bc673ad1b8a2efd86856644a629/contrib/build-linux/sdist/make_sdist.sh#L49

and look at e.g.:
https://cirrus-ci.com/task/6247100135505920?logs=build#L31
```
💬 INFO:  done.
total 8216
drwxr-xr-x 2 user user    4096 Jul  4 12:43 .
drwxr-xr-x 9 user user    4096 Jul  4 12:43 ..
-rw-r--r-- 1 user user 8401137 Jul  4 12:43 Electrum-4.3.0a0.tar.gz
c733079846670b507289b2049125ee76491905b72d87373c4e0fb6bbaa41378e  /tmp/cirrus-ci-build/contrib/build-linux/sdist/../../../dist/Electrum-4.3.0a0.tar.gz
```

I suppose you should branch on `$OMIT_UNCLEAN_FILES` in `make_sdist.sh` to rename the file to e.g. `Electrum-sourceonly-4.3.0a0.tar.gz`.
However, it is not clear to me how exactly to do this. We are calling `python3 setup.py sdist` - I don't think there's an arg we could pass to customise the name. Maybe we can determine what the output name is and simply rename the file. (but even then, as that would not be atomic, note that there can be headaches with the sourceonly tar.gz overwriting the non-sourceonly one...)

Reply:
Oh yeah, whoops.

I tried doing a gross hack, but this also changed the hash of the file. Do we want for `Electrum-sourceonly-4.3.0a0.tar.gz` to contain the directory `Electrum-4.3.0a0/` or `Electrum-sourceonly-4.3.0a0/`? And is the version important?

If the only thing that's to be changed is the filename, I suppose you could just make it emit the output tarball into a different directory? Like:

```s_o -> a/Electrum-4.3.0a0.tar.gz
normal -> Electrum-4.3.0a0.tar.gz
a/Electrum-4.3.0a0.tar.gz -> Electrum-sourceonly-4.3.0a0.tar.gz.```

This isn't atomic, but it shouldn't have overwriting issues at least.

Reply:
> Do we want for `Electrum-sourceonly-4.3.0a0.tar.gz` to contain the directory `Electrum-4.3.0a0/` or `Electrum-sourceonly-4.3.0a0/`?

I don't mind, either is ok.

> And is the version important?

At first I did not understand what you mean by that but then I looked at your "gross hack" in https://github.com/spesmilo/electrum/pull/7594/commits/9edcc427320cfe7babf285d89ad6c9c53b30ef9a ... :P
I've just tested a build, and while it works, I am getting warnings:
```
/home/user/.local/lib/python3.8/site-packages/setuptools/dist.py:530: UserWarning: The version specified ('sourceonly-4.3.0a0') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  warnings.warn(
/home/user/.local/lib/python3.8/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: sourceonly-4.3.0a0 is an invalid version and will not be supported in a future release
  warnings.warn(
```
So let's not do this.

> If the only thing that's to be changed is the filename, I suppose you could just make it emit the output tarball into a different directory? [...]
> This isn't atomic, but it shouldn't have overwriting issues at least.

That sounds fine. Better than hacking the `version` arg...
I think we can use the `--dist-dir` arg, e.g.:
```
python3 setup.py --quiet sdist --format=gztar --dist-dir=dist/sourceonly/
```
And then rename and move the file, like you said. I guess also delete the `dist/sourceonly/` folder after.

Reply:
Sure. I used the folder name `_sourceonly` instead, to clarify that it's meant to be ephemeral. (I guess I could've gone harder on putting this inside of a variable and stuff, but IMO it would've harmed readability more than it'd have helped.)